### PR TITLE
Navigator Refactoring

### PR DESCRIPTION
**Pull request by CebotariM**

### CODE REVIEW

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/WeatherCodesMap.java [hunk 1]
```diff
@@ -9,8 +9,8 @@
 
 public class WeatherCodesMap {
     private Map<String, List<Integer>> map;
-    private int iconResIndex;
-    private int bckgResIndex;
+    private final int iconResIndex;
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/WeatherCodesMap.java]: there is no reason to assign this variable with value in constructor, in this case is better to use constant instead just final variable that is initialized in  constructor. the same for the variable below.

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/navigator/Navigator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        this.sentBundle = new Bundle();
+    }
+
+    public void navigate() {
+        intent.putExtra(ARGS_KEY, sentBundle);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        context.startActivity(intent);
+    }
+
+    public Navigator toClass(Class target) {
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/navigator/Navigator.java]: could you please try to use more meaningful name for this method.

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/navigator/Navigator.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    private static final String ARGS_KEY = "HISTORY_ARGS";
+    private static final String TIMESTAMP_KEY = "TIMESTAMP_KEY";
+    private static final String CITY_ID_KEY = "CITY_ID_KEY";
+    private Context context;
+    private Bundle sentBundle;
+    private Intent intent;
+
+    public Navigator(Context context) {
+        this.context = context;
+        this.sentBundle = new Bundle();
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/navigator/Navigator.java:18]: unfortunately this class is untestable be careful please.

**Review by MaximCeban — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/WeatherCodesMap.java [hunk 4]
```diff
@@ -9,8 +9,8 @@
 
 public class WeatherCodesMap {
     private Map<String, List<Integer>> map;
-    private int iconResIndex;
-    private int bckgResIndex;
+    private static final int iconResIndex = 0;
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/WeatherCodesMap.java]: http://www.oracle.com/technetwork/java/codeconventions-135099.html

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/sharedpreferences/SharedPrefs.java [hunk 5]
```diff
@@ -13,13 +11,12 @@
     private SharedPreferences sharedPref;
     private SharedPreferences.Editor editor;
 
-    public SharedPrefs(Context context) {
-        sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
-        editor = sharedPref.edit();
+    public SharedPrefs(SharedPreferences preferences) {
+        this.sharedPref = preferences;
+        this.editor = preferences.edit();
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/sharedpreferences/SharedPrefs.java]: there is no reason to store editor, you can call it in each method. please avoid storing link to unused class.

**Review by MaximCeban — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/navigator/Navigator.java (hunk 3 above)
Comment (CebotariM) [reply app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/navigator/Navigator.java:18]: The class uses native Android Packages (Intent) and cannot be tested

**Review by MaximCeban — APPROVED** [review]

**Pull request merged**